### PR TITLE
[public-api-server] Simple gRPC server

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Extends public-api-server with a simple gRPC server. The server doesn't have any services any starts up. This is intended to move us along and allow us to iterate, rather than be a complete and final result.

Graceful shutdown of servers is outside of the scope of this PR and will be tackled in subsequent ones.

## Related Issue(s)

Part of https://github.com/gitpod-io/gitpod/issues/9229

## How to test

`cd components/public-api-server && go run main.go`
```
Serving HTTP :9000
Serving gRPC [::]:9001
^CReceived termination signal, shutting down.
```

## Release Notes

```release-note
NONE
```

## Documentation

NONE

### CODE REVIEW

File: components/public-api-server/main.go [hunk 1]
```diff
@@ -6,15 +6,66 @@ package main
...
-	}))
+	errors := make(chan error)
+	defer close(errors)
+
+	go func() {
+		addr := ":9000"
+		fmt.Println("Serving HTTP", addr)
+		err := http.ListenAndServe(addr, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, _ = w.Write([]byte(`hello world\n`))
+		}))
```
Comment (andrew-farries) [inline components/public-api-server/main.go:27]: Do we still need this http server in addition to the GRPC server below?

File: components/public-api-server/main.go [hunk 2]
```diff
@@ -6,15 +6,66 @@ package main
...
+		if err != nil {
+			fmt.Println("Failed to server on", addr, err)
```
Comment (andrew-farries) [inline components/public-api-server/main.go:29]: 'failed to start'

**Review by andrew-farries — CHANGES_REQUESTED** [review]

File: components/public-api-server/main.go (hunk 1 above)
Comment (easyCZ) [reply components/public-api-server/main.go:27]: Yes. We'll need to serve metrics, ready/health endpoints etc

**Review by andrew-farries — APPROVED** [review]

File: components/public-api-server/main.go (hunk 2 above)
Comment (easyCZ) [reply components/public-api-server/main.go]: pushed a fix, thanks!

**Review by andrew-farries — APPROVED** [review]

**Pull request merged**